### PR TITLE
Implement Exclude List Row Actions

### LINKED ISSUE #1754: FF: Add List (Implement for Exclude Table)
Once the Add Include List modal is working for Include, implement for Exclude list.

TBD: what are the differences to note

### LINKED ISSUE #1757: FF Map backend response to Include/Exclude Tables
The Include and Exclude table data design has changed a little and will need to be reimplemented, as well as the endpoint to gather data from.

The inclusion of the 'include all' will be in a separate story.

Each row will have an enable toggle and a couple buttons, but for this particular story we just want to make sure the row data for of each list type is showing up correctly and ready to be wired up.

@bcb37 can you attach an example of the data fetch response that we should be building this table(s) from now?

Latest design:

Image

Image

### LINKED ISSUE #1775: FF: Implement the Edit Button for the "Include/Exclude Lists" Table Rows
Clicking the "Edit" button on the "Include/Exclude Lists" table rows should open the "Edit Include/Exclude List" modal, pre-filled with the row's field values. The "Save" button should remain disabled until the form is modified. Upon clicking the "Save" button after making edits, the modal should close, and the updated values should be reflected in the corresponding table row.

Feature Flag Details

Feature Flag - Edit Include List

### LINKED ISSUE #1776: FF: Implement the Delete Button for the "Include/Exclude Lists" Table Rows
Clicking the "Delete" button on the "Include/Exclude Lists" table rows should open the "Delete Include/Exclude List" modal to confirm the deletion. The message in the modal should include the corresponding list's name. Upon clicking the "Delete" button in the modal, the modal should close, and the corresponding row should be deleted from the table.

Feature Flag Details

Delete Include List

### LINKED ISSUE #1818: FF: Map Exclusion List data to Exclusions table
Map Exclusion List data to Exclusions table

Exactly the same as Include list.

### LINKED ISSUE #1819: FF: Wire up Edit modal for Exclusion List row button
Wire up Edit modal for Exclusion List row button

Essentially same as include list, but will be launched with.

```
export enum UPSERT_PRIVATE_SEGMENT_LIST_ACTION {
  ADD_FLAG_INCLUDE_LIST = 'add_flag_include',
  EDIT_FLAG_INCLUDE_LIST = 'edit_flag_include',
  ADD_FLAG_EXCLUDE_LIST = 'add_flag_exclude',
  EDIT_FLAG_EXCLUDE_LIST = 'edit_flag_exclude', <---- this one
  ADD_SEGMENT_LIST = 'add_segment_list',
  EDIT_SEGMENT_LIST = 'edit_segment_list',
}
```

### LINKED ISSUE #1820: FF: Wire up delete button in exclusion row
This will work the same as include list's delete button functionality, but the message in the modal will be different, which is configured in common-dialog-service.

### PR DESCRIPTION
**Pull request by zackcl**
This PR implements Edit/Delete actions for rows in the Exclude List table on the Feature Flag details page. 

Resolves #1757, resolves #1775, resolves #1776, resolves #1818, resolves #1819, resolves #1820, resolves #1754

### CODE REVIEW

File: frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.selectors.ts [hunk 1]
```diff
@@ -53,7 +53,23 @@ export const selectSelectedFeatureFlag = createSelector(
     // be very defensive here to make sure routerState is correct
     const flagId = routerState?.state?.params?.flagId;
     if (flagId) {
-      return featureFlagState.entities[flagId];
+      const featureFlag = featureFlagState.entities[flagId];
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.selectors.ts:56]: This selector should be kept as simple as possible, it's a very widely used and just needs to access the flag, we already have selectors for the lists that derive from this base selector.

File: frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.selectors.ts [hunk 2]
```diff
@@ -136,19 +152,23 @@ export const selectFeatureFlagInclusions = createSelector(
...
-      .sort((a, b) => new Date(a.segment.createdAt).getTime() - new Date(b.segment.createdAt).getTime())
-      .map((inclusion) => {
-        return {
-          segment: inclusion.segment,
-          listType: inclusion.listType,
-          enabled: inclusion.enabled,
-        };
-      });
+    return featureFlag.featureFlagSegmentInclusion.map((inclusion) => ({
+      segment: inclusion.segment,
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.selectors.ts:156]: why was the sorting stuff moved? It makes more sense here than in the selectSelectedFeatureFlag right?

**Comment by danoswaltCL** [comment]
Just a couple minor comments, I think this should be good. I hope you found working with the `UpsertPrivateSegmentListModal` to be easy breezy.

**Comment by zackcl** [comment]
Thank you, I resolved your comments regarding sorting (I reverted the last commit). I think that makes sense since `selectSelectedFeatureFlag` is called frequently. I just thought it would be cleaner to handle the sorting in one place.

Ideally, I think sorting shouldn't be necessary because it seems that the Postgres database already contains the `feature_flag_segment_inclusion` data in the correct order (when I execute `SELECT * FROM "feature_flag_segment_inclusion";`). However, the order seems different when we actually fetch the data from the backend. I'm not sure if this is expected behavior from Postgres or TypeORM.

Here's the relevant issue I created: https://github.com/CarnegieLearningWeb/UpGrade/issues/1827

Please feel free to approve this PR.

**Pull request merged**